x-pack/plugins/security_solution/server/endpoint/lib/policy/license_watch.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Subscription } from 'rxjs';
+
+import {
+  KibanaRequest,
+  Logger,
+  SavedObjectsClientContract,
+  SavedObjectsServiceStart,
+} from 'src/core/server';
+import { PackagePolicy, PACKAGE_POLICY_SAVED_OBJECT_TYPE } from '../../../../../fleet/common';
+import { PackagePolicyServiceInterface } from '../../../../../fleet/server';
+import { ILicense } from '../../../../../licensing/common/types';
+import {
+  isEndpointPolicyValidForLicense,
+  unsetPolicyFeaturesAboveLicenseLevel,
+} from '../../../../common/license/policy_config';
+import { licenseService } from '../../../lib/license/license';
+
+export class PolicyWatcher {
```

Comment:
`PolicyWatcher` confused me a little because it did not mention that it's really a license watcher. Maybe `LicenseWatcher` would be more appropriate? 

(ps. I can see this being used by other things in the future that might not be Policy related)

Reply:
(two hard problems joke here)

This is in `server/endpoint/lib/policy`, and it's job is to watch _license_ changes, in order to keep _policy_ up to date. I'm open to better names. `LicenseWatcher` is a possibility, but note that our other class, `LicenseService` is copy-pasted from about 4 other plugins that have the same class. But some of them call that `LicenseWatcher`. Wanted to prevent that confusion.

Also I do agree we may want other side-effects to license changes that may not be policy related. But those should create their own subscription via `LicenseService` observable. Not use this. Observables are designed to have anyone subscribe who also needs it, not collectively boggarting off of one sub.